src/Symfony/Component/Cache/DependencyInjection/CacheCollectorPass.php

```diff
@@ -46,29 +46,41 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        $collectorDefinition = $container->getDefinition($this->dataCollectorCacheId);
         foreach ($container->findTaggedServiceIds($this->cachePoolTag) as $id => $attributes) {
-            $definition = $container->getDefinition($id);
-            if ($definition->isAbstract()) {
-                continue;
-            }
+            $this->addToCollector($id, $container);
 
-            $recorder = new Definition(is_subclass_of($definition->getClass(), TagAwareAdapterInterface::class) ? TraceableTagAwareAdapter::class : TraceableAdapter::class);
-            $recorder->setTags($definition->getTags());
-            if (!$definition->isPublic() || !$definition->isPrivate()) {
-                $recorder->setPublic($definition->isPublic());
+            foreach ($attributes as $attribute) {
```

Comment:
do we really need to `foreach`? Isn't look at `$attributes[0]` enough?

Reply:
Indeed, a foreach is not really needed; removed.